source/jormungandr/tests/kirin_realtime_tests.py

```diff
@@ -1918,6 +1918,70 @@ def test_update_trip_with_physical_mode(self):
         assert pt_response['physical_modes'][0]['name'] == 'Tramway'
 
 
+@dataset(MAIN_ROUTING_TEST_SETTING)
+class TestKirinAddTripWithHeadSign(MockKirinDisruptionsFixture):
+    def test_add_trip_with_headsign(self):
+        """
+        1. send a disruption with a headsign to add a trip
+        2. check that headsign is present in journey.section.display_informations
+        """
+        disruption_query = 'disruptions?_current_datetime={dt}'.format(dt='20120614T080000')
+        disruptions_before = self.query_region(disruption_query)
+        nb_disruptions_before = len(disruptions_before['disruptions'])
+
+        # New disruption, a new trip with 2 stop_times in realtime
+        self.send_mock(
+            "additional-trip",
+            "20120614",
+            'added',
+            [
+                UpdatedStopTime(
+                    "stop_point:stopC",
+                    arrival_delay=0,
+                    departure_delay=0,
+                    is_added=True,
+                    arrival=tstamp("20120614T080100"),
+                    departure=tstamp("20120614T080100"),
+                    message='on time',
+                ),
+                UpdatedStopTime(
+                    "stop_point:stopB",
+                    arrival_delay=0,
+                    departure_delay=0,
+                    is_added=True,
+                    arrival=tstamp("20120614T080102"),
+                    departure=tstamp("20120614T080102"),
+                ),
+            ],
+            disruption_id='new_trip',
+            effect='additional_service',
+            headsign='trip_headsign',
+        )
+
+        # Check that a new disruption is added
+        disruptions_after = self.query_region(disruption_query)
+        assert nb_disruptions_before + 1 == len(disruptions_after['disruptions'])
+
+        C_B_query = (
+            "journeys?from={f}&to={to}&data_freshness=realtime&"
+            "datetime={dt}&_current_datetime={dt}".format(
+                f='stop_point:stopC', to='stop_point:stopB', dt='20120614T080000'
+            )
+        )
+
+        # Check that a PT journey exists with first physical_mode in the NTFS('Tramway')
+        response = self.query_region(C_B_query)
+        assert has_the_disruption(response, 'new_trip')
+        self.is_valid_journey_response(response, C_B_query)
+        assert len(response['journeys']) == 2
+        pt_journey = response['journeys'][0]
+        assert 'non_pt_walking' not in pt_journey['tags']
+        assert pt_journey['status'] == 'ADDITIONAL_SERVICE'
+        assert pt_journey['sections'][0]['data_freshness'] == 'realtime'
+        assert pt_journey['sections'][0]['display_informations']['commercial_mode'] == 'additional service'
+        assert pt_journey['sections'][0]['display_informations']['headsign'] == 'trip_headsign'
```

Comment:
Do you really need to test all the things other than the `headsign` ? 
It looks like copy/paste from another test and I don't really see the point on testing multiple time the same thing.

Reply:
You are right, i should have deleted some unnecessary lines